sys/include/net/ipv4/hdr.h

```diff
@@ -0,0 +1,195 @@
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                       Source Address                          |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                    Destination Address                        |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                    Options                    |    Padding    |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc791#section-3.1">
+ *              RFC 791, section 3.1
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    /**
+     * @brief Version and Internet Header Length.
+     *
+     * @details The version are the 4 most significant bits and the Internet Header Length
+     * the 4 next bit (see above).
+     *
+     * This module provides helper functions to set, get, and check these
+     * fields accordingly:
+     * * ipv4_hdr_set_version()
+     * * ipv4_hdr_get_version()
+     * * ipv4_hdr_set_ihl()
+     * * ipv4_hdr_get_ihl()
+     */
+    uint8_t v_ih;
+    uint8_t ts;             /**< type of service of packet*/
+    network_uint16_t tl;    /**< total length of the datagram */
+    network_uint16_t id;    /**< identification value of packet */
+    /**
+     * @brief version control flags and Fragment Offset.
+     *
+     * @details The flags are the 3 most significant bits, and the remaining 13 bits are the fragment offfset
+     *
+     * This module provides helper functions to set, get, and check these
+     * fields accordingly:
+     * * ipv4_hdr_set_flags()
+     * * ipv4_hdr_get_flags()
+     * * ipv4_hdr_set_fo()
+     * * ipv4_hdr_get_fo()
+     */
+    network_uint16_t fl_fo;
+    uint8_t ttl;            /**< time to live for this packet */
+    uint8_t protocol;       /**< protocol of this packet */
+    network_uint16_t csum;  /**< checksum of this packet */
+    ipv4_addr_t src;        /**< source address of this packet */
+    ipv4_addr_t dst;        /**< destination address of this packet */
+} ipv4_hdr_t;
+
+
+/**
+ * @brief   Sets the version field of @p hdr to 6
+ *
+ * @param[out] hdr  Pointer to an IPv4 header.
+ */
+static inline void ipv4_hdr_set_version(ipv4_hdr_t *hdr)
+{
+    hdr->v_ih &= 0x0f;
+    hdr->v_ih |= 0x40;
+}
+
+/**
+ * @brief   Gets the value of the version field of @p hdr
+ *
+ * @param[in] hdr   Pointer to an IPv4 header.
+ *
+ * @return  Value of the version field of @p hdr.
+ */
+static inline uint8_t ipv4_hdr_get_version(ipv4_hdr_t *hdr)
+{
+    return ((hdr->v_ih) >> 4);
+}
+
+/**
+ * @brief   Sets the Internet Header Length field of @p hdr
+ *
+ * @param[out] hdr  Pointer to an IPv4 header.
+ * @param[in] ihl  Size in bytes of the Internet Header Length (including padding)
+ */
+static inline void ipv4_hdr_set_ihl(ipv4_hdr_t *hdr, uint8_t ihl)
```

Comment:
With this type `ihl` can only be 255 at max (224 if we consider that it has to be divisable by 32). Is this (or a smaller value) somewhere specified as a maximum? Otherwise, I would prefer to use `uint16_t` here.